### PR TITLE
issue-1030

### PR DESCRIPTION
**Pull request by bell87**
added a call to filterAvailableChildren function which will allow page to me marked as complete when articles/blocks/components have _isAvailable flag set to false. 

Related to https://github.com/adaptlearning/adapt_framework/issues/1030 and framework pr: https://github.com/adaptlearning/adapt_framework/pull/1102

### CODE REVIEW

**Comment by dennis-learningpool** [comment]
+1

**Comment by barry-learningpool** [comment]
+1

File: js/completionCalculations.js [hunk 1]
```diff
@@ -110,9 +111,32 @@ define([
         });
     }
 
+    function unavailableInHierarchy(parents) {
+        if (parents.length > 0) {
```
Comment (brian-learningpool) [inline js/completionCalculations.js]: Is it actually possible to have a child with no parents?  If so, I think there's a missing `else` and `return` in this code.

**Comment by oliverfoster** [comment]
+1

**Comment by moloko** [comment]
+1

**Comment by brian-learningpool** [comment]
+1

**Pull request merged**